csbuild/_utils/decorators.py

```diff
@@ -234,7 +234,7 @@ def _wrap(*args, **kwargs):
 						# pick the correct matching passed-in argument
 						if i < len(args):
 							elem = args[i]
-						elif name in kwargs:
+						elif name in kwargs: # pylint: disable=bad-option-value,consider-using-get
```

Comment:
I actually like pylint's suggestion here, too.

```
if i < len(args):
    elem = args[i]
else:
    elem = kwargs.get(name, NOT_SET)
```
And then the initializer above can be removed.